commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitTest.java

```diff
@@ -0,0 +1,58 @@
+package org.infinispan.commons.test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.testng.ITestResult;
+
+public class PolarionJUnitTest {
+
+   public enum Status {SUCCESS, FLAKY, FAILURE, ERROR, SKIPPED}
+
+   final String name;
+   final String clazz;
+   final List<Throwable> failures = new ArrayList<>();
```

Comment:
The `AtomicInteger/Long` suggests that this class may be updated concurrently. Should we replace `ArrayList` with `CopyOnWriteArrayList`?

Reply:
Good point. As we're frequently writing, wouldn't using `Collections.synchronizeList()` be more efficient as I can't imagine contention will be an issue.

Reply:
I'm not concerned about perf here. I'm ok using `Collections.synchronizeList` here too.